cpp/src/phonenumbers/phonenumbermatcher.cc

```diff
@@ -421,12 +422,26 @@ PhoneNumberMatcher::PhoneNumberMatcher(const string& text,
       max_tries_(numeric_limits<int>::max()),
       state_(NOT_READY),
       last_match_(NULL),
-      search_index_(0) {
+      search_index_(0),
+      isInputInCleanUTF8_(true) {
+  isInputInCleanUTF8_ =  CheckInputUTF8OrNot();
 }
 
+
 PhoneNumberMatcher::~PhoneNumberMatcher() {
 }
 
+// static
+bool PhoneNumberMatcher::CheckInputUTF8OrNot() {
+  // Input should contain only UTF-8 characters.
+  UnicodeText number_as_unicode;
+  number_as_unicode.PointToUTF8(text_.c_str(), text_.size());
+  if (number_as_unicode.UTF8WasValid()) {
```

Comment:
better: return number_as_unicode.UTF8WasValid(); instead of the if/else?

Reply:
+1. Done.